src/gui/FileBrowser.cpp

```diff
@@ -992,21 +1026,96 @@ bool FileBrowserTreeWidget::openInNewSampleTrack(FileItem* item)
 
 
 
+void FileBrowserTreeWidget::openDirectory(Directory* directory) {
+   QDesktopServices::openUrl(QUrl::fromLocalFile(directory->fullName()));
+}
+
+#if !defined(_WIN32) && !defined(__APPLE__)
+
+bool supportsSelectOption(const QString &fileManager) {
+	if (selectOptionCache.contains(fileManager)) {
+		return selectOptionCache[fileManager];
+	}
+
+	QProcess process;
+	process.start(fileManager, {"--help"});
+
+	// Ensure we don't block forever
+	if (!process.waitForFinished(1000)) {
+		selectOptionCache[fileManager] = false;
+		return false;
+	}
+
+	// Capture both stdout and stderr (some apps print help to stderr)
+	QString output = QString::fromUtf8(process.readAllStandardOutput() + process.readAllStandardError());
```

Comment:
Out of an overabundance of caution, this may need to be `QString::fromLocal8Bit`, since platforms can use some strange encodings by default.  This is certainly the case for Windows, but I think most *nixes are safe from this, so the exciting code might be preferred.